src/Microsoft.AspNet.Mvc.Abstractions/ModelBinding/ModelBindingContext.cs

```diff
@@ -90,6 +100,11 @@ public ModelBindingContext()
         public OperationBindingContext OperationBindingContext { get; set; }
 
         /// <summary>
+        /// Gets or sets the name of the current field being bound.
+        /// </summary>
+        public string FieldName { get; set; }
```

Comment:
This needs `[param: NotNull]` for the setter. Same for `ModelName`, `ModelState`, and most of the existing properties.